### PR TITLE
1.5.8

### PR DESCRIPTION
**Pull request by skupr-anaconda**

### CODE REVIEW

File: recipe/meta.yaml [hunk 1]
```diff
@@ -45,14 +43,18 @@ test:
     - nuitka.tools
     - nuitka.tree
     - nuitka.utils
-  requires:
-    - pip
+  #requires:
+  #  - pip
```
Comment (danpetry) [inline recipe/meta.yaml]: I think you could remove the commented code (but leave the comment explaining reasoning)

File: recipe/meta.yaml [hunk 2]
```diff
@@ -47,12 +46,15 @@ test:
     - nuitka.utils
   requires:
     - pip
+    - {{ compiler('c') }}  # [linux]
```
Comment (danpetry) [inline recipe/meta.yaml:49]: After adding this, you then removed the compilation test `nuitka Empty.py`, so is this still required?

File: recipe/meta.yaml [hunk 3]
```diff
@@ -47,12 +46,15 @@ test:
...
   commands:
     - pip check
     - nuitka --help
+    - nuitka-run --help
+    - python -m nuitka --version
```
Comment (danpetry) [inline recipe/meta.yaml:54]: There is also a test suite upstream. What about running this?

File: recipe/meta.yaml [hunk 4]
```diff
@@ -1,39 +1,37 @@
...
+  number: 0
+  script: {{ PYTHON }} -m pip install . --no-deps --no-deps --no-build-isolation -vv
   entry_points:
     - nuitka = nuitka.__main__:main
     - nuitka-run = nuitka.__main__:main
 
 requirements:
   host:
     - python
+    - libpython-static  # [not win]
```
[the remark this answers is not in the document]
Comment (skupr-anaconda) [reply recipe/meta.yaml:22]: I guess _no_ because every libpython-static has its own Python version

File: recipe/meta.yaml (hunk 3 above)
Comment (skupr-anaconda) [reply recipe/meta.yaml:54]: It's not working because of a frozen prompt without choosing a variant Yes/no.

File: recipe/meta.yaml [hunk 5]
```diff
@@ -1,39 +1,38 @@
...
     - pip
     - setuptools
```
[the remark this answers is not in the document]
Comment (skupr-anaconda) [reply recipe/meta.yaml:24]: For `host`, we use the latest `setuptools` without pinning

[the remark this answers is not in the document]
Comment (skupr-anaconda) [reply recipe/meta.yaml]: You were right. It's not required for `host`. I've removed it

File: recipe/meta.yaml (hunk 4 above)
[the remark this answers is not in the document]
Comment (skupr-anaconda) [reply recipe/meta.yaml]: You were right. It's not required for `host`. I've removed it

File: recipe/meta.yaml (hunk 2 above)
Comment (skupr-anaconda) [reply recipe/meta.yaml]: I returned the example

File: recipe/meta.yaml (hunk 3 above)
Comment (skupr-anaconda) [reply recipe/meta.yaml:52]: The test suite has a pretty complicated settings https://github.com/Nuitka/Nuitka/blob/1.5.8/nuitka/tools/testing/run_nuitka_tests/__main__.py

File: recipe/meta.yaml [hunk 6]
```diff
@@ -1,39 +1,37 @@
...
     - wheel
   run:
     - python
+    - libpython-static  # [not win]
```
[the remark this answers is not in the document]
Comment (danpetry) [reply recipe/meta.yaml:27]: ah, got it

File: recipe/meta.yaml (hunk 5 above)
[the remark this answers is not in the document]
Comment (danpetry) [reply recipe/meta.yaml]: still might be nice to state this explicitly imo, but a minor nit

File: recipe/meta.yaml (hunk 3 above)
Comment (danpetry) [reply recipe/meta.yaml:52]: yeah at first glance it looked pretty complicated/non-automatable. No worries, we can leave it out.

File: recipe/meta.yaml (hunk 4 above)
[the remark this answers is not in the document]
Comment (danpetry) [reply recipe/meta.yaml]: yeah, there's a pinning between them done in the python feedstock

File: recipe/meta.yaml (hunk 3 above)
Comment (skupr-anaconda) [reply recipe/meta.yaml:53]: I've fixed the test compilation by disabling ccache in `nuitka --show-scons --run --verbose --disable-ccache Empty.py` it bypassed the downloading step

Comment (boldorider4) [reply recipe/meta.yaml:53]: Is it not possible to automate the test suite by wrapping the suite in a script that is able to choose variant Yes/No?

**Review by boldorider4 — APPROVED** [review]
